src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentMetadataAsSourceFileProvider.cs

```diff
@@ -55,7 +55,15 @@ internal sealed class PdbSourceDocumentMetadataAsSourceFileProvider : IMetadataA
             _logger = logger;
         }
 
-        public async Task<MetadataAsSourceFile?> GetGeneratedFileAsync(Workspace workspace, Project project, ISymbol symbol, bool signaturesOnly, MetadataAsSourceOptions options, string tempPath, CancellationToken cancellationToken)
+        public async Task<MetadataAsSourceFile?> GetGeneratedFileAsync(
+            MetadataAsSourceWorkspace metadataWorkspace,
+            Workspace sourceWorkspace,
```

Comment:
naming these distinctly to keep it clear which is which.  `Workspace workspace` is now `MetadataAsSourceWorkspace metadataWorkspace`,  and `Workspace sourceWorkspace` is new and corresponds to `Project sourceProject`.